extensions/vercast/src/index.tsx

```diff
@@ -102,3 +123,85 @@ function Main(): JSX.Element {
     </List>
   )
 }
+
+
+function UpdateEnvironmentVariable(props: { projectId: string, projectName: string }): JSX.Element {
+  // Get props values
+  const projectId = props.projectId;
+  const projectName = props.projectName;
+
+  // Get preference values
+  const token = String(preferences.token.value)
+  if (token.length !== 24) {
+    showToast(ToastStyle.Failure, 'Invalid token detected')
+    throw new Error('Invalid token length detected')
+  }
+
+  // Setup useState objects
+  const [environmentVariables, setEnvironmentVariables] = useState<EnvironmentVariable[]>()
+
+  // On form submit function
+  const { pop } = useNavigation();
+  async function handleSubmit(values: FormValues) {
+    if (!validateForm(values)) {
+      return;
+    }
+    const envConf = JSON.parse(values.env_conf)
+    const updatedEnvVariable = await updateEnvironmentVariable(projectId, envConf.envId, envConf.envKey, values.env_value)
+
+    if(!updatedEnvVariable || !updatedEnvVariable.key){
+      showToast(ToastStyle.Failure, "Couldn't update variable");
+    }else{
+      showToast(ToastStyle.Success, updatedEnvVariable.key+" was updated!");
+      pop();
+    }
+  }
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const fetchedEnvironmentVariables = await fetchEnvironmentVariables(
+        projectId
+      )     
+      setEnvironmentVariables(fetchedEnvironmentVariables)
+    }
+    fetchData()
+  }, [projectId])
+  return (
+    <Form 
+      navigationTitle={projectName+" ->  Environment Variables"}
+      isLoading={!environmentVariables} 
+      actions={
+        <ActionPanel>
+          <ActionPanel.Section title="Environment Variable">
+            <SubmitFormAction 
+              title="Update Variable"
+              icon={Icon.Pencil}
+              onSubmit={handleSubmit}
+               />
+            </ActionPanel.Section>
+        </ActionPanel>
+      }>
+      <Form.Dropdown 
+        id="env_conf" 
+        title="Environment Variable">
+          {environmentVariables?.map((ev) => {
+            const randomID = randomUUID()
+            return (
+              <Form.Dropdown.Item 
+                value={JSON.stringify({envId : ev.id, envKey : ev.key})} 
+                title={ev.key} />
+            )
+          })}
+      </Form.Dropdown>
+      <Form.TextArea id="env_value" title="New Value" />
+    </Form>
+  )
+}
+
+function validateForm(values: FormValues): boolean {
+  if (!values.env_value) {
+    showToast(ToastStyle.Failure, "Please set new value");
+    return false;
+  }
+  return true;
+}
```

Comment:
If this is only being called once then it could just be an anonymous function call.

Reply:
Hello @gleich 👋
Not sure how to do that?

Reply:
do something like this where the function is being called:

```js
if (!(values) => {
  if (!values.env_value) {
    showToast(ToastStyle.Failure, "Please set new value");
    return false;
  }
  return true;
}) {
  return;
}
```

Reply:
or you could store the function `validateForm` inside of `UpdateEnvironmentVariable`.

Reply:
I went for the latter and added `validateForm` inside of `UpdateEnvironmentVariable`.